### DISCUSSION TITLE
How to use web api method POST in BIRT

### DISCUSSION THREAD

**Discussion by raviprsd18** [eclipse-birt/birt#1728] [Q&A] [answered]
Hello Team,

I am currently exploring the possibility of using the web API POST method with BIRT APIs. Could any one kindly guide me on how to achieve this?

**For example, I am looking to implement the following code snippet in BIRT using its APIs:**

var result = "";

const xhr = new XMLHttpRequest();
xhr.open("POST", "https://jsonplaceholder.typicode.com/todos");
xhr.setRequestHeader("Content-Type", "application/json; charset=UTF-8");
const body = JSON.stringify({
  userId: 1,
  title: "Fix my bugs",
  completed: false
});
xhr.onload = function() {
  if (xhr.readyState == 4 && xhr.status == 201) {
    result = JSON.parse(xhr.responseText);
  } else {
    result = xhr.status;
  }
};
xhr.send(body);

If you could provide a sample code or any relevant resources, it would be greatly appreciated.
Thank you in advance for your time and assistance.

**Comment by speckyspooky** [comment]
Long story short, you cannot use "xhr"-object because this is an object which comes from the JS-engine from the browser.
The core of BIRT works with ECMAScript and there isn't all options from JS-engines implemented because EMCAScript is the server-oriented version of JavaScript.

But yes, it is possible to use REST-apis and here you have to use Java-methods.
I created an API-report based on a BIRT report and implemented also the response handling of XML-messages an so on.

**Reply by raviprsd18** [reply]
@speckyspooky thanks for the reply.

Can I get any sample or resource to do hands on.

**Reply by raviprsd18** [reply]
@speckyspooky gentle remainder

**Reply by speckyspooky** [reply]
I read your post and will take a look into my reports.

**Reply by speckyspooky** [reply, marked as answer]
...here you get my 2 demos, my be this can help you.
The report includes also the references from my used research sources.

### Execution of a web-request: REST & response status handling
[rest_post_request.zip](https://github.com/user-attachments/files/15814528/rest_post_request.zip)

### Execution of response-parsing "xml-response"
[xml_parser_javax.zip](https://github.com/user-attachments/files/15814530/xml_parser_javax.zip)

**Reply by raviprsd18** [reply]
@speckyspooky thanks, will keep you updated.

**Reply by raviprsd18** [reply]
@speckyspooky thanks for the solution. Working as expected.
